source/common/log_sink/log_obj_store.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include <vector>
+
+#include "common/common/assert.h"
+
+//#include "envoy/access_log/access_log.h"
+
+namespace Envoy {
+namespace LogTransport {
+
+/**
+ * Provide a generic wrapper for a block of multiple log objects (of any type).
```

Comment:
Are we batching before placing on the conveyor? What are the tradeoffs with batching at the other end of the conveyor, at the consumer? Seems if we had lots of producers sending messages, the consumer might be able to do a better job at batching..